src/libs/ReportUtils.js

```diff
@@ -3277,20 +3277,34 @@ function chatIncludesChronos(report) {
 /**
  * Can only flag if:
  *
- * - It was written by someone else
+ * - It was written by someone else and isn't a whisper
+ * - It's a welcome message whisper
  * - It's an ADDCOMMENT that is not an attachment
  *
  * @param {Object} reportAction
  * @param {number} reportID
  * @returns {Boolean}
  */
 function canFlagReportAction(reportAction, reportID) {
+    const report = getReport(reportID);
+    const isCurrentUserAction = reportAction.actorAccountID === currentUserAccountID;
+
+    if (ReportActionsUtils.isWhisperAction(reportAction)) {
+        // Allow flagging welcome message whispers as they can be set by any room creator
+        if (lodashGet(reportAction, 'originalMessage.html') === report.welcomeMessage && !isCurrentUserAction) {
```

Comment:
I tested this, the welcome message is always plain text so this comparison is safe